Mathlib/Logic/Equiv/Defs.lean

```diff
@@ -301,6 +301,15 @@ theorem symm_bijective : Function.Bijective (Equiv.symm : (α ≃ β) → β ≃
 
 @[simp] theorem self_trans_symm (e : α ≃ β) : e.trans e.symm = Equiv.refl α := ext <| by simp
 
+theorem trans_eq_refl_iff_eq_symm (f : α ≃ β) (g : β ≃ α) :
+    f.trans g = Equiv.refl α ↔ f = g.symm := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · ext x
+    apply g.injective
+    rw [← trans_apply, h, refl_apply, apply_symm_apply]
+  · rw [h, ← self_trans_symm]
+    rfl
```

Comment:
Here's a golf if you put this at line 497 after `symm_comp_eq`:
```
theorem trans_eq_refl_iff_eq_symm (f : α ≃ β) (g : β ≃ α) :
    f.trans g = Equiv.refl α ↔ f = g.symm := by
  rw [← Equiv.coe_inj, coe_trans, coe_refl, ← eq_symm_comp, comp_id, Equiv.coe_inj]
```

Reply:
This is nice, thank you.